core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -0,0 +1,226 @@
+
+  /** Updates the element at the index, or throws an exeption if none exists */
+  def updatedUnsafe(i: Int, a: A):
+      NonEmptyVector[A] = NonEmptyVector(toVector.updated(i, a))
+
+  def head: A = toVector.head
+
+  def tail: Vector[A] = toVector.tail
+
+  /**
+   * remove elements not matching the predicate
+   */
+  def filter(f: A => Boolean): Vector[A] = toVector.filter(f)
+
+  /**
+   * Append another NonEmptyVector to this
+   */
+  def concat(other: NonEmptyVector[A]): NonEmptyVector[A] = NonEmptyVector(toVector ++ other.toVector)
+
+  /**
+    * Alias for concat
+    */
+  def ++(other: NonEmptyVector[A]): NonEmptyVector[A] = concat(other)
+
+  /**
+   * Append another Vector to this
+   */
+  def concat(other: Vector[A]): NonEmptyVector[A] = NonEmptyVector(toVector ++ other)
+
+  /**
+    * Alias for concat
+    */
+  def ++(other: Vector[A]): NonEmptyVector[A] = concat(other)
+
+  /**
+   * find the first element matching the predicate, if one exists
+   */
+  def find(f: A => Boolean): Option[A] = toVector.find(f)
+
+  /**
+   * Check whether at least one element satisfies the predicate.
+   */
+  def exists(f: A => Boolean): Boolean = toVector.exists(f)
+
+  /**
+   * Check whether all elements satisfy the predicate.
+   */
+  def forall(f: A => Boolean): Boolean = toVector.forall(f)
+
+  /**
+   * Left-associative fold using f.
+   */
+  def foldLeft[B](b: B)(f: (B, A) => B): B =
+    toVector.foldLeft(b)(f)
+
+  /**
+   * Right-associative fold using f.
+   */
+  def foldRight[B](lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+    Foldable[Vector].foldRight(toVector, lb)(f)
+
+  /**
+    * Applies f to all the elements
+    */
+  def map[B](f: A => B): NonEmptyVector[B] =
+    NonEmptyVector(toVector.map(f))
+
+  /**
+    *  Applies f to all elements and combines the result
+    */
+  def flatMap[B](f: A => NonEmptyVector[B]): NonEmptyVector[B] =
+    NonEmptyVector(toVector.flatMap(a => f(a).toVector))
+
+  /**
+    * Left-associative reduce using f.
+    */
+  def reduceLeft(f: (A, A) => A): A =
+    tail.foldLeft(head)(f)
+
+  /**
+    * Reduce using the Semigroup of A
+    */
+  def reduce(implicit S: Semigroup[A]): A =
+    S.combineAllOption(toVector).get
+
+  /**
+   * Typesafe equality operator.
+   *
+   * This method is similar to == except that it only allows two
+   * NonEmptyVector[A] values to be compared to each other, and uses
+   * equality provided by Eq[_] instances, rather than using the
+   * universal equality provided by .equals.
+   */
+  def ===(that: NonEmptyVector[A])(implicit A: Eq[A]): Boolean = Eq[Vector[A]].eqv(toVector, that.toVector)
+
+  /**
+   * Typesafe stringification method.
+   *
+   * This method is similar to .toString except that it stringifies
+   * values according to Show[_] instances, rather than using the
+   * universal .toString method.
+   */
+  def show(implicit A: Show[A]): String =
+    s"NonEmptyVector(${Show[Vector[A]].show(toVector)})"
```

Comment:
I think this will give us something like `NonEmptyVector(Vector(1, 2, 3))` for `NonEmptyVector(1, 2, 3)`.
Maybe we could do `"NonEmpty" + Show[Vector[A]].show(toVector)` which should give us `NonEmptyVector(1,2,3)` ?